controllers/helm.go

```diff
@@ -109,6 +89,51 @@ func mergeHelmConfigs(meta *ectypes.ReleaseMetadata, in *v1beta1.Installation) *
 	return combinedConfigs
 }
 
+// update the 'admin-console' and 'embedded-cluster-operator' charts to add cluster ID, binary name, and airgap status
+func updateInfraChartsFromInstall(ctx context.Context, in *v1beta1.Installation, charts k0sv1beta1.ChartsSettings) k0sv1beta1.ChartsSettings {
+	log := ctrl.LoggerFrom(ctx)
+
+	if in == nil {
+		return charts
+	}
+
+	for i, chart := range charts {
+		if chart.Name == "admin-console" {
+			// admin-console has "embeddedClusterID" and "isAirgap" as dynamic values
```

Comment:
The idea is that we don't add any new protected fields, and if there are new things we need to override on an ongoing basis we do that in here instead of in embedded cluster